public/app/features/annotations/standardAnnotationSupport.ts

```diff
@@ -0,0 +1,188 @@
+import {
+  DataFrame,
+  AnnotationQuery,
+  AnnotationSupport,
+  PanelData,
+  transformDataFrame,
+  FieldType,
+  Field,
+  KeyValue,
+  AnnotationEvent,
+  AnnotationEventMappings,
+  getFieldDisplayName,
+  AnnotationEventFieldSource,
+} from '@grafana/data';
+
+import isString from 'lodash/isString';
+
+export const standardAnnotationSupport: AnnotationSupport = {
+  /**
+   * Assume the stored value is standard model.
+   */
+  prepareAnnotation: (json: any) => {
+    if (isString(json?.query)) {
+      const { query, ...rest } = json;
+      return {
+        ...rest,
+        target: {
+          query,
+        },
+        mappings: {},
+      };
+    }
+    return json as AnnotationQuery;
+  },
+
+  /**
+   * Convert the stored JSON model and environment to a standard datasource query object.
+   * This query will be executed in the datasource and the results converted into events.
+   * Returning an undefined result will quietly skip query execution
+   */
+  prepareQuery: (anno: AnnotationQuery) => anno.target,
+
+  /**
+   * When the standard frame > event processing is insufficient, this allows explicit control of the mappings
+   */
+  processEvents: (anno: AnnotationQuery, data: DataFrame) => {
+    return getAnnotationsFromFrame(data, anno.mappings);
+  },
+};
+
+/**
+ * Flatten all panel data into a single frame
+ */
+export function singleFrameFromPanelData(rsp: PanelData): DataFrame | undefined {
+  if (!rsp?.series?.length) {
+    return undefined;
+  }
+  if (rsp.series.length === 1) {
+    return rsp.series[0];
+  }
+
+  return transformDataFrame(
+    [
+      {
+        id: 'seriesToColumns',
+        options: { byField: 'Time' },
```

Comment:
Ah, using the same assumption (name: Time) in graph panel work. Do you consider this safe? I don't think we gurantee the name of the time field, do we?

Reply:
oh right -- i copied this from you ;)  it would be better to have a function that does this for us in a consistent way.

Reply:
think it's better to use merge transform as it more automatically merges data (seriesToColumns also has a big limitation in that it does not handle duplicate rows for the same time value)

Reply:
Not sure about merge TBH. Doesn't it work in a way that if you have two series:
A: [Time, Value]
B: [Time, Value, Value1]

it will result in :
C: [Time,Value,Value1]

@mckn correct my if I'm wrong ^^

Reply:
@dprokop yes - but only if the values of Time, Value are exactly the same. Then the merge will consider those "lines" as being the same and hence merge them to one.